be/src/olap/tablet.cpp

```diff
@@ -206,6 +206,24 @@ OLAPStatus Tablet::revise_tablet_meta(const TabletMeta& tablet_meta,
         _tablet_meta = new_tablet_meta;
     } while (0);
 
+    for (auto& version : versions_to_delete) {
+        auto it = _rs_version_map.find(version);
+        _rs_version_map.erase(it);
+    }
+
+    for (auto& rs_meta : rowsets_to_clone) {
+        Version version = { rs_meta->start_version(), rs_meta->end_version() };
+        RowsetSharedPtr rowset(new AlphaRowset(&_schema, _tablet_path, _data_dir, rs_meta));
+        _rs_version_map[version] = rowset;
```

Comment:
I think rowset should be added to the map after init and you should return failure if init failed.

Reply:
OK